React/Views/RCTComponentData.m

```diff
@@ -395,42 +396,47 @@ - (void)setProps:(NSDictionary<NSString *, id> *)props forShadowView:(RCTShadowV
     }
   }
 #pragma clang diagnostic pop
-
-  unsigned int count = 0;
-  NSMutableDictionary *propTypes = [NSMutableDictionary new];
-  Method *methods = class_copyMethodList(object_getClass(_managerClass), &count);
-  for (unsigned int i = 0; i < count; i++) {
-    SEL selector = method_getName(methods[i]);
-    const char *selectorName = sel_getName(selector);
-    if (strncmp(selectorName, "propConfig", strlen("propConfig")) != 0) {
-      continue;
-    }
-
-    // We need to handle both propConfig_* and propConfigShadow_* methods
-    const char *underscorePos = strchr(selectorName + strlen("propConfig"), '_');
-    if (!underscorePos) {
-      continue;
-    }
-
-    NSString *name = @(underscorePos + 1);
-    NSString *type = ((NSArray<NSString *> *(*)(id, SEL))objc_msgSend)(_managerClass, selector)[0];
-    if (RCT_DEBUG && propTypes[name] && ![propTypes[name] isEqualToString:type]) {
-      RCTLogError(@"Property '%@' of component '%@' redefined from '%@' "
-                  "to '%@'", name, _name, propTypes[name], type);
-    }
-
-    if ([type isEqualToString:@"RCTBubblingEventBlock"]) {
-      [bubblingEvents addObject:RCTNormalizeInputEventName(name)];
-      propTypes[name] = @"BOOL";
-    } else if ([type isEqualToString:@"RCTDirectEventBlock"]) {
-      [directEvents addObject:RCTNormalizeInputEventName(name)];
-      propTypes[name] = @"BOOL";
-    } else {
-      propTypes[name] = type;
+  
+  Class superClass = _managerClass;
+  while (superClass && superClass != [RCTViewManager class]) {
+    if ([superClass isSubclassOfClass:[RCTViewManager class]]) {
```

Comment:
Yeah, it worked before even without iterating over RCTViewManager! So, even it it does now, it is not intentional. I would try to optimize this iteration a bit by using some ObjC runtime functions (because we can and because we care 😄 ), but I want to have conceptual consensus before.